### PR TITLE
Support installing requirements without root

### PR DESCRIPTION
**Pull request by simonklb**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

Some users like to install their requirements in a user directory without root instead of globally. I found myself parsing the `get-requirements.yaml` file and installing things manually. This makes it useful for me (and others that have the same preference).

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

**Review by robinAwallace — APPROVED** [review]

File: bin/install-requirements.bash [hunk 1]
```diff
@@ -6,4 +6,9 @@ export ANSIBLE_STDOUT_CALLBACK=yaml
 
 echo "This script will install all requirements"
 echo "Enter your password to start the install"
-ansible-playbook -e 'ansible_python_interpreter=/usr/bin/python3' --ask-become-pass --connection local --inventory 127.0.0.1, roles/get-requirements.yaml
+
+if [ "${#}" -ge 2 ] && [ "${2}" = "--user" ]; then
+  ansible-playbook -e 'ansible_python_interpreter=/usr/bin/python3' --ask-become-pass --extra-vars install_path="${HOME}/bin" roles/get-requirements.yaml
```
Comment (aarnq) [inline bin/install-requirements.bash:11]: Could you make this configurable? :pray: I put stuff in `.local/bin`.

Comment (simonklb) [reply bin/install-requirements.bash]: Does this work for you? https://github.com/elastisys/compliantkubernetes-apps/pull/2142/commits/703190a596f0573264c554404a26ae1eea6e4cc3

Comment (aarnq) [reply bin/install-requirements.bash]: Yeah :+1:

File: roles/get-requirements.yaml [hunk 2]
```diff
@@ -45,11 +46,10 @@
         dest: "{{ install_path }}/yajsv"
         version_flag: "-v"
 
-  connection: local
-  become: yes
-  become_user: root
```
Comment (aarnq) [inline roles/get-requirements.yaml:50]: Will the rootfull still work without become for the install steps? :thinking:

Comment (simonklb) [reply roles/get-requirements.yaml:50]: As long as the current user has write permission on `$CK8S_INSTALL_PATH` it should only be required for the packages which are installed using `apt`. That's why I moved it to the `apt` task and added the `--become` flag to the `ansible-playbook` execution when you don't use `--user`.

From what I can tell it works as intended:
```shell
$ sudo mkdir /reqtest
$ CK8S_INSTALL_PATH=/reqtest ./bin/ck8s install-requirements
...
$ ls -al /reqtest/
total 271992
drwxr-xr-x  2 root root       4096 maj 17 16:36 .
drwxr-xr-x 21 root root       4096 maj 17 16:33 ..
-rwxrwxr-x  1 1001 docker 51228672 dec 13 22:13 helm
-rwxrwxr-x  1 1001 docker 76030104 feb 18 11:50 helmfile
-rwxrwxr-x  1 root root    3953824 maj 17 16:36 jq
-rwxr-xr-x  1 root root   49889280 jan 17 15:07 kubectl
-rwxr-xr-x  1 root root   48828416 jan 17 15:07 kubectl-convert
-rwxrwxr-x  1 root root   22511616 maj 17 16:36 sops
-rwxrwxr-x  1 root root   10022635 maj 17 16:36 yajsv
-rwxrwxr-x  1 root root    6076668 maj 17 16:36 yq
-rwxrwxr-x  1 root root    9928856 maj 17 16:36 yq4
```

Comment (simonklb) [reply roles/get-requirements.yaml:50]: And of course the `--user` cases:

```shell
$ mkdir /tmp/reqtest
$ CK8S_INSTALL_PATH=/tmp/reqtest ./bin/ck8s install-requirements --user
...
$ ls -al /tmp/reqtest/
total 272052
drwxrwxr-x  2 simon simon     4096 maj 17 16:42 .
drwxrwxrwt 20 root  root     81920 maj 17 16:42 ..
-rwxrwxr-x  1 simon simon 51228672 dec 13 22:13 helm
-rwxrwxr-x  1 simon simon 76030104 feb 18 11:50 helmfile
-rwxrwxr-x  1 simon simon  3953824 maj 17 16:41 jq
-rwxr-xr-x  1 simon simon 49889280 jan 17 15:07 kubectl
-rwxr-xr-x  1 simon simon 48828416 jan 17 15:07 kubectl-convert
-rwxrwxr-x  1 simon simon 22511616 maj 17 16:42 sops
-rwxrwxr-x  1 simon simon 10022635 maj 17 16:42 yajsv
-rwxrwxr-x  1 simon simon  6076668 maj 17 16:42 yq
-rwxrwxr-x  1 simon simon  9928856 maj 17 16:42 yq4
```

```shell
$ CK8S_INSTALL_PATH=/reqtest ./bin/ck8s install-requirements --user
...
TASK [Get Kubectl] *************************************************************************************************************
fatal: [localhost]: FAILED! => changed=false
  dest: /reqtest
  extract_results:
    cmd:
    - /usr/bin/tar
    - --extract
    - -C
    - /reqtest
    - -z
    - --show-transformed-names
    - --strip-components=3
    - -f
    - /home/simon/.ansible/tmp/ansible-tmp-1715956778.285057-1546110-68490811102477/kubernetes-client-linux-amd64c0s5oi7x.tar.gz
    err: |-
      /usr/bin/tar: kubectl: Cannot open: Permission denied
      /usr/bin/tar: kubectl-convert: Cannot open: Permission denied
      /usr/bin/tar: Exiting with failure status due to previous errors
    out: ''
    rc: 2
  gid: 0
  group: root
  handler: TgzArchive
  mode: '0755'
  msg: failed to unpack /home/simon/.ansible/tmp/ansible-tmp-1715956778.285057-1546110-68490811102477/kubernetes-client-linux-amd64c0s5oi7x.tar.gz to /reqtest
  owner: root
  size: 4096
  src: /home/simon/.ansible/tmp/ansible-tmp-1715956778.285057-1546110-68490811102477/kubernetes-client-linux-amd64c0s5oi7x.tar.gz
  state: directory
  uid: 0
```

Comment (aarnq) [reply roles/get-requirements.yaml:50]: Ah, I missed that you added the `--become --become-user root` flags for rootful install.

**Review by aarnq — APPROVED** [review]

**Pull request merged**